app/src/main/java/org/simple/clinic/home/patients/PatientsScreenController.kt

```diff
@@ -218,9 +222,13 @@ class PatientsScreenController @Inject constructor(
         .listen()
         .filter { it is ShowAppUpdate }
 
+    val now = LocalDate.now(utcClock)
+
     return Observables
         .combineLatest(screenCreated, availableUpdate)
         .map { (_, update) -> update }
+        .filter { appUpdateDialogShownAtPref.get().atZone(utcClock.zone).toLocalDate().isBefore(now) }
```

Comment:
Since the feature requires us to check calendar dates, we should be using the user clock to generate the `LocalDate` instead of the utc clock.

I tried to do this and clean up the chain a little bit.

```kotlin
return Observables
        .combineLatest(screenCreated, availableUpdate)
        .map { (_, update) -> update }
        .filter { hasADayPassedSinceAppUpdateWasLastShown() }
        .doOnNext { appUpdateDialogShownAtPref.set(Instant.now(utcClock)) }
        .map { Ui::showAppUpdateDialog }

private fun hasADayPassedSinceAppUpdateWasLastShown(): Boolean {
    val today = LocalDate.now(userClock)
    val dateOnWhichAppUpdateWasLastShown = appUpdateDialogShownAtPref.get().toLocalDateAtZone(userClock.zone)

    return dateOnWhichAppUpdateWasLastShown.isBefore(today)
}
```

I also changed the saving of the app update shown at preference do a side effect block.

Reply:
Good point. I will change this.